src/index.js

```diff
@@ -101,20 +103,22 @@ class Libp2p extends EventEmitter {
    */
   static async create (options) {
     if (options.peerId) {
+      // @ts-ignore 'Libp2pOptions & CreateOptions' is not assignable to 'Libp2pOptions & constructorOptions'
```

Comment:
maybe these `Libp2pOptions`, `CreateOptions` and `constructorOptions` types need to be refactored to be more composable ?

Reply:
Do you have any suggestion? I think that the issue is that one says that the peerId is required while the other says it is optional. This causes:

```
Argument of type 'Libp2pOptions & CreateOptions' is not assignable to parameter of type 'Libp2pOptions & constructorOptions'.
  Type 'Libp2pOptions & CreateOptions' is not assignable to type 'constructorOptions'.
    Types of property 'peerId' are incompatible.
      Type 'PeerId | undefined' is not assignable to type 'PeerId'.
        Type 'undefined' is not assignable to type 'PeerId'
```

Reply:
i have been dealing with this type of stuff lately, specifically with constructor options that have options props but internally we have defaults for all or some.

So what im doing now is:
```js
class Libp2p {
  
  constructor(options: Libp2pOptions) {
    const opts: Required<Libp2pOptions> = {...options, ...defaults} 
 }
}
```
This avoids having to truthy check everywhere, but also makes all options required so kinda forces you to have defaults for everything which might sounds troublesome but i think its good as a general rule.

Reply:
Ok, we might be able to move into that in the future once we revamp libp2p config, but not now. This would mean we need to have libp2p modules as dependencies here.